### DISCUSSION TITLE
V7. How to make form dirty programmatically?

### DISCUSSION THREAD

**Discussion by michael-land** [react-hook-form/react-hook-form#4522] [Q&A]
My use case is that I have a multi-steps form, each step is individual form component.

When the user go back to previous step, the form read previous value from the cache, and reset the form. I also want to make the form dirty, any idea how to do that?  Or is there anyway to pass default from states when call useForm({})?

**Comment by bluebill1049** [comment]
i think you will have to use `setValue` with `shouldDirty`
